server/src/main/java/org/elasticsearch/action/admin/cluster/stats/AnalysisStats.java

```diff
@@ -47,6 +47,16 @@ public final class AnalysisStats implements ToXContentFragment, Writeable {
 
     private static final Set<String> SYNONYM_FILTER_TYPES = Set.of("synonym", "synonym_graph");
 
+    // Maps the synonyms token filter configurations to the stats keys used
+    static final Map<String, String> SYNONYM_STATS_KEYS_FOR_CONFIG = Map.of(
+        "synonyms",
+        "inline",
+        "synonyms_set",
```

Comment:
Is this for backwards compatibility? Could `synonyms_set` and `synonyms_path` be removed from the map?

Reply:
Sorry, I don't get the question 😢  

The goal of this map is to perform the translation between what we find in the synonyms token filters config (synonyms_set, synonyms_path, synonyms) to what the actual keys in stats will look like.

It's not about BWC, but used to translate these keys.

Hopefully is that what was confusing you? 🤞